src/test/java/games/strategy/engine/lobby/server/login/RsaAuthenticatorTest.java

```diff
@@ -33,23 +34,17 @@ public void testCanProcess() {
   }
 
   @Test
-  public void testPublicKeysAreExpungedAfterLookup() {
-    final Cache<String, PrivateKey> rsaKeyCache = CacheBuilder.newBuilder().build();
-    final RsaAuthenticator rsaAuthenticator = new RsaAuthenticator(rsaKeyCache);
-    final String publicKey = "something";
+  public void testRoundTripPassword() {
+    final String password = "password";
     final Map<String, String> challenge = new HashMap<>();
-    challenge.put(RsaAuthenticator.RSA_PUBLIC_KEY, publicKey);
-
-    assertFalse("There are no public keys in a pristine instance, so we expect key lookup to fail",
-        rsaAuthenticator.getPrivateKey(challenge).isPresent());
-
-    rsaKeyCache.put(publicKey, mockPrivateKey);
+    final Map<String, String> response = new HashMap<>();
+    @SuppressWarnings("unchecked")
+    final Function<String, String> action = mock(Function.class);
```

Comment:
Usimg spy on a do nothing function would get rid of a potential unchecked operation.

Reply:
I'm pretty sure spying works on anonymous classes, while mocks don't.

Reply:
Good suggestion.  I didn't realize there was a `spy(T)` overload, as I'm only used to seeing `spy(Class<T>)`.  In general, I dislike `spy()` because it requires the target class to be non-`final`, and I'm a big believer in Item 17 from _Effective Java_ (Design and document for inheritance or else prohibit it).

I changed the line in question to

```java
final Function<String, String> action = spy(p -> null);
```

which allowed me to remove the `@SuppressWarnings` annotation, as you noted.  However, the anonymous class generated by the lambda is apparently `final` as Mockito complains when running the test:

```
org.mockito.exceptions.base.MockitoException: 
Cannot mock/spy class games.strategy.engine.lobby.server.login.RsaAuthenticatorTest$$Lambda$1/537548559
Mockito cannot mock/spy because :
 - final class
```

😞 

If I were using this mock in more than one place, I'd try harder to get rid of that `@SuppressWarnings` by

1. extracting it to a field with a `@Mock` annotation, or
1. extracting a custom non-generic type, e.g. `interface DecryptPasswordAction extends Function<String, String> {}`, and mocking that instead, e.g. `final Function<String, String> action = mock(DecryptPasswordAction.class);`.